hydra-main/src/main/java/com/addthis/hydra/job/spawn/Spawn.java

```diff
@@ -1388,6 +1397,7 @@ private void updateJobDependencies(String jobId) {
         if ((job == null) || (job.getParameters() == null)) {
             return dataSources;
         }
+        // Maybe
```

Comment:
Given that `parameters` is a non-threadsafe collection that is exposed directly via a getter, I think that some protection is necessary. There is nothing to stop someone from calling job.getParameters.add(something) at an inopportune moment. 

Changing the getter to return a copy rather than the actual list might be a more lightweight solution